apis/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -75,7 +75,10 @@ func (host *BareMetalHost) validateChanges(old *BareMetalHost) []error {
 		errs = append(errs, err...)
 	}
 
-	if old.Spec.BMC.Address != "" && host.Spec.BMC.Address != old.Spec.BMC.Address {
+	if old.Spec.BMC.Address != "" &&
+		host.Spec.BMC.Address != old.Spec.BMC.Address &&
+		host.Status.OperationalStatus != OperationalStatusDetached &&
+		host.Status.Provisioning.State != StateRegistering {
 		errs = append(errs, errors.New("BMC address can not be changed once it is set"))
```

Comment:
not sure this message is still valid as we can now update the BMC address

Reply:
I have corrected the message now